src/plugins/auto/src/schedule.cpp

```diff
@@ -0,0 +1,307 @@
+    SoAsyncInferRequest request_to_share_blobs_with;
+    auto request_id = m_request_id.fetch_add(1);
+    if (m_context->m_bind_buffer) {
+        size_t sum = 0;
+        for (const auto& device : m_context->m_device_priorities_initial) {
+            auto& dev_requests = m_workerrequests[device.device_name];
+            if ((request_id - sum) <  dev_requests.size()) {
+                request_to_share_blobs_with = dev_requests.at(request_id - sum).m_inferrequest;
+                INFO_RUN([&]() {
+                    std::lock_guard<std::mutex> lock(m_dev_infer_mutex);
+                    m_dev_infer.insert(std::make_pair(request_to_share_blobs_with._ptr, &dev_requests.at(request_id - sum)));
+                });
+                break;
+            }
+            sum += dev_requests.size();
+        }
+        if (!request_to_share_blobs_with) {
+            OPENVINO_THROW("binder mode does not allow oversubsciption of infer requests, please use optimal infer request");
+        }
+    } else if (m_passthrough_exenet) {
+        request_to_share_blobs_with = {m_passthrough_exenet->create_infer_request(), m_passthrough_exenet._so};
+    }
+    return std::make_shared<InferRequest>(
+        std::static_pointer_cast<const CompiledModel>(compiled_model), request_to_share_blobs_with);
+}
+
+void Schedule::run(ov::threading::Task pipeline_task) {
+    schedule_to_worker_inferrequest(std::move(pipeline_task), m_this_preferred_devicename);
+}
+
+bool Schedule::run_pipeline_task(ov::threading::Task& pipeline_task,
+    NotBusyPriorityWorkerRequests& idle_workerrequests,
+    const DeviceName& preferred_device) {
+    WorkerInferRequest* worker_request_ptr = nullptr;
+    std::pair<int, WorkerInferRequest*> worker;
+    if (idle_workerrequests.try_pop(worker)) {
+        worker_request_ptr = worker.second;
+        IdleGuard<NotBusyPriorityWorkerRequests> idle_guard{worker_request_ptr, idle_workerrequests};
+        m_this_worker_inferrequest = worker_request_ptr;
+        {
+            auto captured_task = std::move(pipeline_task);
+            captured_task();
+        }
+        idle_guard.release();
+        return true;
+    }
+    return false;
+}
+
+void Schedule::generate_workers(const std::string& device, const SoCompiledModel& executable_network) {
+    std::string real_devicename;
+    if (device == "CPU_HELP") {
+        real_devicename = "CPU";
+    } else {
+        real_devicename = device;
+    }
+    auto it_numrequests = deviceChecker().check_and_return_if_device_in_list<DeviceInformation>(real_devicename, m_context->m_device_priorities, true);
+    unsigned int optimal_num = 0;
+    try {
+        optimal_num = executable_network->get_property(ov::optimal_number_of_infer_requests.name()).as<unsigned int>();
+    } catch (const ov::Exception& iie) {
+        OPENVINO_THROW("Every device used with the Multi-Device should support OPTIMAL_NUMBER_OF_INFER_REQUESTS ExecutableNetwork metric",
```

Comment:
Please, correct this message:
- No MULTI_DEVICE I suppose
- No legacy `OPTIMAL_NUMBER_OF_INFER_REQUESTS ` name
- No ExecutableNetwork 
- No Metric